### DISCUSSION TITLE
[Cardano] Doesn't TrustWallet now support Cardano for transaction signing?

### DISCUSSION THREAD

**Discussion by dengpan1** [trustwallet/wallet-core#1351] [Q&A]
How do I sign Cardano transactions?

**Comment by dengpan1** [comment]
I used the JNI way

**Comment by hewigovens** [comment]
Cardano signing is not supported yet

**Reply by Hazhir007** [reply]
is there some document about cardano algorithms for creating wallets and signing transactions? i can not find any in the cardano website.

**Reply by optout21** [reply]
If you care to dig up things from cardano wallet sources:
https://github.com/vacuumlabs/cardano-crypto.js
https://github.com/vacuumlabs/adalite            
https://adalite.io/
https://daedaluswallet.io/

**Reply by Hazhir007** [reply]
thanks

**Reply by Hazhir007** [reply]
this source is awesome it also has implemented signing transactions: https://github.com/CardanoSharp/cardanosharp-wallet

**Comment by Hazhir007** [comment]
can we use the AddressV3.cpp addresses generator to use for signing transactions? because i saw that addresses generating by TW are longer but other libraries for example Emurgo produces shorter addresses and when trying to validate TW addresses with Emurgo they produce exceptions. should i try to create another Address generator?


another question i had was regarding that how to produce hex string of an address for example with emurgo we can get "000743d16cfe3c4fcc0c11c2403bbc10dbc7ecdd4477e053481a368e7a06e2ae44dff6770dc0f4ada3cf4cf2605008e27aecdb332ad349fda7" from "addr_test1qqr585tvlc7ylnqvz8pyqwauzrdu0mxag3m7q56grgmgu7sxu2hyfhlkwuxupa9d5085eunq2qywy7hvmvej456flknswgndm3" which we need to use the hex version in the CBOR structure that is "000743d16cfe3c4fcc0c11c2403bbc10dbc7ecdd4477e053481a368e7a06e2ae44dff6770dc0f4ada3cf4cf2605008e27aecdb332ad349fda7"
but when trying to decode an address such as "addr1s3alxrr29g6s5vmculs3unkrq66r040t3pe0g5hgwwjre8rqqtfw39n97rqyxzq3jtcncknxg4zppvnzwkz9q9cns2nn728yptedesppr0m552" that is produced by TW and using :


```cpp
auto bech = Bech32::decode("addr1s3alxrr29g6s5vmculs3unkrq66r040t3pe0g5hgwwjre8rqqtfw39n97rqyxzq3jtcncknxg4zppvnzwkz9q9cns2nn728yptedesppr0m552");
auto addressbech32 = std::get<0>(bech);
cout << "bech32 address encoded: " << addressbech32 << endl;
```
        
it returns 40 in the console.

**Comment by Hazhir007** [comment]
how to print out the payload part of an address as a hex string in console?, the type is Data i assume
```cpp
auto bech = Bech32::decode("addr1s3alxrr29g6s5vmculs3unkrq66r040t3pe0g5hgwwjre8rqqtfw39n97rqyxzq3jtcncknxg4zppvnzwkz9q9cns2nn728yptedesppr0m552");
auto payloadOfAddress = get<1>(decoded);
```

**Comment by hewigovens** [comment]
Now we supported signing Cardano tx
